django/contrib/admin/templates/admin/base.html

```diff
@@ -59,6 +60,12 @@
                     {% csrf_token %}
                     <button type="submit">{% translate 'Log out' %}</button>
                 </form>
+                <button class="theme-toggle">
+                  <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
+                  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24" class="theme-icon-when-auto"><path d="M0 0h24v24H0z"/><path d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2V4a8 8 0 1 0 0 16z"/></svg>
+                  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24" class="theme-icon-when-dark"><path d="M0 0h24v24H0z"/><path d="M10 7a7 7 0 0 0 12 4.9v.1c0 5.523-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2h.1A6.979 6.979 0 0 0 10 7zm-6 5a8 8 0 0 0 15.062 3.762A9 9 0 0 1 8.238 4.938 7.999 7.999 0 0 0 4 12z"/></svg>
+                  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24" class="theme-icon-when-light"><path d="M0 0h24v24H0z"/><path d="M12 18a6 6 0 1 1 0-12 6 6 0 0 1 0 12zm0-2a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM11 1h2v3h-2V1zm0 19h2v3h-2v-3zM3.515 4.929l1.414-1.414L7.05 5.636 5.636 7.05 3.515 4.93zM16.95 18.364l1.414-1.414 2.121 2.121-1.414 1.414-2.121-2.121zm2.121-14.85l1.414 1.415-2.121 2.121-1.414-1.414 2.121-2.121zM5.636 16.95l1.414 1.414-2.121 2.121-1.414-1.414 2.121-2.121zM23 11v2h-3v-2h3zM4 11v2H1v-2h3z"/></svg>
```

Comment:
Is it worth defining these as symbols, so that we don't have a duplication on the raw SVG data? 🤔

https://css-tricks.com/svg-use-with-external-reference-take-2/

Reply:
I already tried with symbols but with external referenced SVG we can't access directly to shapes. I use them to style the color of icons with the default colors :
```
.theme-toggle svg.theme-icon-when-auto > path:nth-of-type(1),
.theme-toggle svg.theme-icon-when-dark > path:nth-of-type(1),
.theme-toggle svg.theme-icon-when-light > path:nth-of-type(1){
    fill: var(--header-bg);
}

.theme-toggle svg.theme-icon-when-auto > path:nth-of-type(2),
.theme-toggle svg.theme-icon-when-dark > path:nth-of-type(2),
.theme-toggle svg.theme-icon-when-light > path:nth-of-type(2){
    fill: var(--header-link-color);
}
```

Reply:
Can we use Unicode symbols instead? e.g. ☾, ⛭

Reply:
I'm not sure there is one for auto mode, and I have to check for color and how it renders and come back with a more precise answer

Reply:
> I can define these as symbols but the raw svg must be in the same page, I guess, best will be in base.html ...

This is what I had in mind. 🤔

The small amount of markup isn't a problem to include (I don't see it as a big deal at least) but if we can avoid duplicating it, that would be good, otherwise we have a slight maintenance itch ≈forever going forwards.

(Not a biggie if it's not feasible but good to try?)

Reply:
Update: unicode symbols didn't render well, especially the auto mode, I finally go for SVGs symbols. 
I just update css in order to apply color for each icon 🎉